src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/batch/BatchManager.kt

```diff
@@ -3,28 +3,40 @@ package uk.gov.justice.digital.hmpps.prisonertonomisupdate.batch
 import kotlinx.coroutines.runBlocking
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
+import org.springframework.beans.factory.getBeanProvider
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.context.ConfigurableApplicationContext
 import org.springframework.context.event.ContextRefreshedEvent
 import org.springframework.context.event.EventListener
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.activities.ActivitiesReconService
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.DomainEventListener
 
 @ConditionalOnProperty(name = ["batch.enabled"], havingValue = "true")
 @Service
-class BatchController(
+class BatchManager(
   @Value($$"${batch.type}") private val batchType: String,
   private val activitiesReconService: ActivitiesReconService,
 ) {
 
   @EventListener
   fun onApplicationEvent(event: ContextRefreshedEvent) = runBlocking {
+    removeQueueListeners(event)
+
     when (batchType) {
       "ALLOCATION_RECON" -> activitiesReconService.allocationReconciliationReport()
       else -> log.error("Batch type $batchType not supported")
     }
   }.also { event.closeApplication() }
 
+  private fun removeQueueListeners(event: ContextRefreshedEvent) {
+    with(event.applicationContext.autowireCapableBeanFactory) {
+      getBeanProvider<DomainEventListener>().forEach {
+        this.destroyBean(it)
+      }
+    }
+  }
```

Comment:
Remove all queue listeners - seems better than requiring the listeners to add a Conditional to stop consuming when running in batch mode